File: Algorithms/C/prime_number.c
#include <stdio.h>
#include <stdbool.h>

bool is_prime(int n)
{
    if (n <= 1)
    {
        return false;
    }
    if (n <= 3)
    {
        return true;
    }
    if (n % 2 == 0 || n % 3 == 0)
    {
        return false;
    }
    for (int i = 5; i * i <= n; i += 6)
    {
        if (n % i == 0 || n % (i + 2) == 0)
        {
            return false;
        }
    }
    return true;
}

int main()
{
    int n = 7;
    if (is_prime(n))
    {
        printf("%d is a prime number\n", n);
    }
    else
    {
        printf("%d is not a prime number\n", n);
    }
    return 0;
}

File: Algorithms/Python/prime_number.py
from __future__ import annotations


def is_prime(number: int) -> bool:
    if number <= 1:
        return False

    if number <= 3:
        return True

    if number % 2 == 0 or number % 3 == 0:
        return False

    i = 5

    while i**2 <= number:
        if number % i == 0 or number % (i + 2) == 0:
            return False
        i += 6

    return True


if __name__ == "__main__":
    print(is_prime(11))  # True
    print(is_prime(12))  # False
